src/main/java/uk/gov/service/notify/JsonUtils.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.service.notify;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+
+public class JsonUtils {
+    public static Map<String, Object> jsonToMap(JSONObject json) throws JSONException {
+        Map<String, Object> retMap = new HashMap<>();
+
+        if(json != JSONObject.NULL) {
+            retMap = toMap(json);
+        }
+        return retMap;
+    }
+
+    private static Map<String, Object> toMap(JSONObject object) throws JSONException {
+        Map<String, Object> map = new HashMap<>();
+
+        Iterator<String> keysItr = object.keys();
+        while(keysItr.hasNext()) {
+            String key = keysItr.next();
+            Object value = object.get(key);
+
+            if(value instanceof JSONArray) {
+                value = toList((JSONArray) value);
+            }
+
+            else if(value instanceof JSONObject) {
+                value = toMap((JSONObject) value);
+            }
+            map.put(key, value);
+        }
+        return map;
+    }
+
+    public static List<Object> jsonToList(JSONArray json) throws JSONException {
+        List<Object> retList = new ArrayList<>();
+
+        if(json != JSONObject.NULL) {
+            retList = toList(json);
+        }
+        return retList;
+    }
+
+    private static List<Object> toList(JSONArray array) throws JSONException {
+        List<Object> list = new ArrayList<>();
+        for(int i = 0; i < array.length(); i++) {
```

Comment:
if you are using Java 8 then you can use

`JSONArray array = ...;`

`array.forEach(item -> {
    JSONObject obj = (JSONObject) item;
    parse(obj);
});`

Reply:
JsonUtils refactored and rewritten using Java 8